eos/s-decays/k-to-pi-l-nu.cc

```diff
@@ -0,0 +1,536 @@
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/form-factors/form-factors.hh>
+#include <eos/maths/integrate.hh>
+#include <eos/maths/power-of.hh>
+#include <eos/models/model.hh>
+#include <eos/s-decays/k-to-pi-l-nu.hh>
+#include <eos/utils/destringify.hh>
+#include <eos/utils/kinematic.hh>
+#include <eos/utils/options-impl.hh>
+#include <eos/utils/private_implementation_pattern-impl.hh>
+
+#include <map>
+#include <string>
+
+namespace eos
+{
+    namespace k_to_pi_l_nu
+    {
+        struct Amplitudes
+        {
+                // helicity amplitudes, cf. [DDS:2014A] eqs. 13-14
+                complex<double> h_0;
+                complex<double> h_t;
+                complex<double> h_S;
+                complex<double> h_T;
+                complex<double> h_tS;
+                double          v;
+                double          p;
+                double          NF;
+        };
+    } // namespace k_to_pi_l_nu
+
+    template <> struct Implementation<KToPiLeptonNeutrino>
+    {
+            std::shared_ptr<Model> model;
+
+            Parameters parameters;
+
+            RestrictedOption opt_K;
+
+            UsedParameter m_K;
+
+            UsedParameter tau_K;
+
+            UsedParameter m_pi;
+
+            LeptonFlavorOption opt_l;
+
+            UsedParameter m_l;
+
+            UsedParameter g_fermi;
+
+            UsedParameter hbar;
+
+            const double isospin_factor;
+
+            UsedParameter mu;
+
+            static const std::vector<OptionSpecification> options;
+
+            GSL::QAGS::Config int_config;
+
+            BooleanOption opt_cp_conjugate;
+
+            std::shared_ptr<FormFactors<VacuumToPP>> form_factors;
+
+            Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+                model(Model::make(o.get("model"_ok, "SM"), p, o)),
+                parameters(p),
+                opt_K(o, options, "K"_ok),
+                m_K(p["mass::" + opt_K.value()], u),
+                tau_K(p["life_time::" + opt_K.value()], u),
+                m_pi(p["mass::pi^" + std::string(opt_K.value() == "K_u" ? "0" : "-")], u),
+                opt_l(o, options, "l"_ok),
+                m_l(p["mass::" + opt_l.str()], u),
+                g_fermi(p["WET::G_Fermi"], u),
+                hbar(p["QM::hbar"], u),
+                isospin_factor(opt_K.value() == "K_L" ? +1.0 : -1.0), // K_L -> pi+ vs K_S -> pi+ and K- -> pi^0
```

Comment:
I'm pretty sure there should be a sqrt(2) factor somewhere, on the basis the K- BR is wrong